test/net-test.js

```diff
@@ -0,0 +1,2218 @@
+        // of the receiver of the message, it is the local.
+        const pkt = new packets.VersionPacket({remote: local});
+
+        await pool.handleVersion(peer, pkt);
+
+        assert(called);
+      });
+    });
+
+    describe('handleAddr', function() {
+      it('will add addrs to hosts list', async () => {
+        const network = Network.get('regtest');
+
+        const pool = new Pool({
+          logger: Logger.global,
+          chain: {
+            network,
+            options: {
+              checkpoints: true
+            },
+            on: () => {},
+            state: {
+              hasWitness: () => true
+            }
+          }
+        });
+
+        const peer = Peer.fromOptions(pool.options);
+
+        const items = [
+          // Routable and has required services
+          new NetAddress({
+            host: '5.19.5.127',
+            port: 8334,
+            services: 0 | services.NETWORK | services.WITNESS,
+            time: 1558405603
+          }),
+          // Routable and missing services
+          new NetAddress({
+            host: '5.29.139.120',
+            port: 8335,
+            services: 0 | services.NETWORK,
+            time: 1558405603
+          }),
+          // Not routable
+          new NetAddress({
+            host: '127.0.0.3',
+            port: 8333,
+            services: 0 | services.NETWORK | services.WITNESS,
+            time: 1558405602
+          })
+        ];
+
+        const pkt = new packets.AddrPacket(items);
+
+        assert.equal(pool.hosts.totalFresh, 0);
+
+        await pool.handleAddr(peer, pkt);
+
+        assert.equal(pool.hosts.totalFresh, 1);
+      });
+    });
+
+    describe('handleInv', function() {
+      const network = Network.get('regtest');
+
+      it('will ban with too many inv', async () => {
+        const pool = new Pool({
+          logger: Logger.global,
+          chain: {network, options: {checkpoints: true}, on: () => {}}
+        });
+
+        let handleTXInv = false;
+        pool.handleTXInv = () => {
+          handleTXInv = true;
+        };
+
+        let increaseBan = false;
+        const peer = Peer.fromOptions(pool.options);
+        peer.increaseBan = (score) => {
+          assert.equal(score, 100);
+          increaseBan = true;
+        };
+
+        const items = [];
+        for (let i = 0; i < common.MAX_INV + 1; i++)
+          items.push(new InvItem(InvItem.types.TX, Buffer.alloc(32, 0x01)));
+
+        const pkt = new packets.InvPacket(items);
+
+        await pool.handleInv(peer, pkt);
+
+        assert(increaseBan);
+        assert(!handleTXInv);
+      });
+
+      it('will handle block inventory', async () => {
+        const pool = new Pool({
+          logger: Logger.global,
+          chain: {network, options: {checkpoints: true}, on: () => {}}
+        });
+
+        const peer = Peer.fromOptions(pool.options);
+
+        let handleBlockInv = false;
+        pool.handleBlockInv = (_peer, blocks) => {
```

Comment:
I _believe_ an `inv` on the network can contain mixed types, but bcoin splits the items up between the two handlers you're testing here. Would it make sense also to cover `pool.handleInv()` with a mixed `inv` ? Could also cover the `invFilter`.

Reply:
Yeah, there is actually quite a lot left to be tested here. Will need additional test cases.